src/main/java/gil/server/controllers/DirectoryController.java

```diff
@@ -0,0 +1,79 @@
+package gil.server.controllers;
+
+import gil.server.http.HTTPProtocol;
+import gil.server.http.Request;
+import gil.server.http.Response;
+import gil.server.router.Routes;
+import java.util.function.BiFunction;
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.MessageFormat;
+
+public class DirectoryController {
+
+    static DirectoryStream.Filter<Path> filter = new DirectoryStream.Filter<Path>() {
+        @Override
+        public boolean accept(Path entry) {
+            return !entry.toFile().isHidden();
+        }
+    };
+
+    static String buildHTMLListItems(Request request) throws ResourceNotFoundException {
+        MessageFormat linkFormat = new MessageFormat("<li><a href=\"{0}\">{1}</a></li>");
+        Integer indexAfterForwardSlash = 1;
+        String uri = request.getURI().substring(indexAfterForwardSlash);
+        StringBuilder listItemBuilder = new StringBuilder();
+
+        try {
+            Files.newDirectoryStream(Paths.get(uri), filter)
+                    .forEach(entry -> {
+                        String link = "/" + entry.toString();
+                        String linkLabel = entry.getFileName().toString();
+
+                        if (entry.toFile().isFile()) {
+                            link = uri.substring(Routes.PUBLIC_FOLDER.length()) + "/" + entry.getFileName().toString();
+                        }
+
+                        listItemBuilder.append(linkFormat.format(new Object[]{link, linkLabel}));
+                    });
+        } catch (IOException e) {
+            throw new ResourceNotFoundException();
+        }
+
+        return listItemBuilder.toString();
+    }
+
+    public static BiFunction<Request, Response, Response> get = (request, response) -> {
+
+        try {
+            String listItems = buildHTMLListItems(request);
+            StringBuilder bodyBuilder = new StringBuilder();
+            bodyBuilder.append("<html>")
```

Comment:
I wonder if it would be better to extract out the HTML building to some service?